app/src/components/AddCredentialSlider.tsx

```diff
@@ -0,0 +1,154 @@
+            padding: 20,
+            shadowOffset: {
+                width: 0,
+                height: 2
+            },
+            shadowOpacity: 0.25,
+            shadowRadius: 4,
+            elevation: 5
+        },
+        drawerTitleText: {
+            ...TextTheme.normal,
+            fontWeight: 'bold',
+            textAlign: 'center',
+            marginVertical: 10
+        },
+        drawerContentText: {
+            ...TextTheme.normal,
+        },
+        drawerRow: {
+            flexDirection: 'row',
+            justifyContent: 'flex-start',
+            alignItems: 'center',
+            marginVertical: 12,
+        },
+        drawerRowItem: {
+            color: ColorPallet.grayscale.black
+        }
+    });
+
+    const deactivateSlider = useCallback(() => {
+        dispatch({
+            type: BCDispatchAction.ADD_CREDENTIAL_PRESSED,
+            payload: [false],
+        })
+    }, [])
+
+    const goToScanScreen = useCallback(() => {
+        deactivateSlider()
+        navigation.getParent()?.navigate(Stacks.ConnectStack, { screen: Screens.Scan })
+    }, [])
+
+
+    const onBCIDPress = useCallback(() => {
+        setWorkflowInFlight(true)
+        startFlow(agent, store, dispatch as React.Dispatch<ReducerAction<any>>, setWorkflowInFlight, t)
+    }, [store])
+
+    useEffect(() => {
+        const animation = Animated.loop(Animated.timing(rotationAnim, timing))
+        animation.reset()
+        if (workflowInFlight) {
+            animation.start()
+        } else {
+            animation.stop()
+        }
+    }, [rotationAnim, workflowInFlight, store.addCredential.addCredentialPressed])
+
+    useEffect(() => {
+        const credentialDefinitionIDs = credentials.map(
+            (c) => c.metadata.data[CredentialMetadataKeys.IndyCredential].credentialDefinitionId as string
+        )
+
+        setShowGetFoundationCredential(showBCIDSelector(credentialDefinitionIDs, canUseLSBCredential))
+
+    }, [credentials, canUseLSBCredential])
+
+    return (
+        <View>
+            <Modal
+                animationType="slide"
+                transparent={true}
+                visible={store.addCredential.addCredentialPressed}
+                onRequestClose={deactivateSlider}
+            >
+                <View style={styles.centeredView}>
+                    <View style={styles.modalView}>
+                        <TouchableOpacity onPress={deactivateSlider}>
+                            <Icon name='window-close' size={35} style={styles.drawerRowItem}></Icon>
+                        </TouchableOpacity>
+                        <Text style={styles.drawerTitleText}>Choose</Text>
+                        {showGetFoundationCredential && (<TouchableOpacity style={styles.drawerRow} disabled={workflowInFlight} onPress={onBCIDPress}>
+
+                            {workflowInFlight ? (
+                                <Animated.View style={[{ transform: [{ rotate: rotation }] }]}>
```

Comment:
Its helpful to organize code like this animation into its own component. Some things can easily be carved out and live on their own. At 150 lines this file is not in bad shape but food for though.

Reply:
Got it, I've factored out the animation code into it's own component